endstate_correction/protocol.py

```diff
@@ -1,14 +1,86 @@
 """Provide functions for the endstate correction workflow."""
 
 
-from openmm.app import Simulation
-import numpy as np
 from dataclasses import dataclass
-from typing import List
+from typing import List, Union
 
+import numpy as np
+import pandas as pd
+from openmm import unit
+from openmm.app import Simulation
 from pymbar import MBAR
 
 
+class BSSProtocol:
+    """This is a dataclass mimicking the BioSimSpace.Protocol."""
+
+    def __init__(
+        self,
+        timestep: Union[int, unit.Quantity],
+        n_integration_steps: int,
+        temperature: Union[float, unit.Quantity],
+        pressure: Union[float, unit.Quantity],
+        report_interval: int,
+        restart_interval: int,
+        rlist: Union[float, unit.Quantity],
+        collision_rate: Union[float, unit.Quantity],
+        switchDistance: Union[float, unit.Quantity],
+        lam: pd.Series,
+        restart: bool,
+    ):
+        """Class for storing the run information
+
+        :param timestep: fs
+        :param n_integration_steps:
+        :param temperature: K
+        :param pressure: atm
+        :param report_interval: The frequency at which energy are recorded (In integration steps).
+        :param restart_interval: The frequency at which frames are recorded (In integration steps).
+        :param rlist: short-range cutoff in nanometers.
+        :param collision_rate: 1/picosecond
+        :param switchDistance: nanometers
+        :param lam: Current lambda
+        :param restart: Whether to reset the velocity or not
+        """
+        if isinstance(timestep, type(unit.femtosecond)):
+            self.timestep = timestep
+        else:
+            self.timestep = timestep * unit.femtosecond
```

Comment:
I'd approach this a bit differently: 
```
if `isinstance(timestep, type(unit.Quantity):
    try:
        timestep.value_in_unit(unit.femtoseconds)
    except Exception as e::
        raise e
    self.timestep = timestep
else:
    self.timestep = timestep * unit.femtosecond
```

Reply:
does this work with 1/unit.femtoseconds as well?

Reply:
I'd think so! It is also of type `unit.Quantity`